HopsanCore/src/ComponentSystem.cpp

```diff
@@ -1871,12 +1871,22 @@ bool ComponentSystem::checkModelBeforeSimulation()
                 break;
             }
 
-            // If a system parameter is used in the component, then erase it from the list
-            std::vector<HString>::iterator itp = std::find(unusedSysParNames.begin(), unusedSysParNames.end(), pCompParameters->at(p)->getValue());
-            if(itp != unusedSysParNames.end())
+            // If a system parameter is used in the component, then erase it from the list.
+            // To match only complete parameter names, prepend and append a space to the expression and replace operators by a space.
+            // Then find parameter names in expression.
+            HString expression = (" " + pCompParameters->at(p)->getValue() + " ")
+                    .replace("+", " ")
+                    .replace("-", " ")
+                    .replace("*", " ")
+                    .replace("/", " ")
+                    .replace("^", " ")
+                    .replace("(", " ")
+                    .replace(")", " ");
+            for(std::vector<HString>::iterator itp = unusedSysParNames.begin(); itp!=unusedSysParNames.end();)
             {
-                unusedSysParNames.erase(itp);
+                (expression.find(" "+(*itp)+" ")!=std::string::npos) ? unusedSysParNames.erase(itp) : itp++;
```

Comment:
I belive erasing from the container (vector) while looping over it is unsafe. I think that iterators may no longer be valid after erase. Does erase return an iterator ? Or is that c++11 (cant use that in hopsancore).

Reply:
True. Made a new commit. Hopefully it will automatically show up here in some way?

Reply:
Yes that should work